tests/integration/einstein/test_operations.py

```diff
@@ -222,3 +223,42 @@ def tearDown(self):
         self.db.project.delete_with_name(PROJECT)
         self.db.close()
         self.good_bmi.shutdown()
+
+
+class TestCopyImage(TestCase):
+    """
+    Creating a flatten copy of an image
+    """
+    @trace
+    def setUp(self):
+        self.db = Database()
+        self.db.project.insert(PROJECT, NETWORK)
+
+        self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
+                            PROJECT)
+        self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
+        self.img2 = 'img2'
+
+    def runTest(self):
+        response = self.good_bmi.copy_image(EXIST_IMG_NAME, PROJECT,
+                                            self.img2)
+        self.assertEqual(response[constants.STATUS_CODE_KEY], 200)
+        images = self.db.image.fetch_images_from_project(PROJECT)
+        exists_image = False
+        for image in images:
+            if self.img2 == image:
+                exists_image = True
+                break
+        self.assertTrue(exists_image)
+        with ceph.RBD(_cfg.fs,
+                      _cfg.iscsi.password) as fs:
+            img_id = self.good_bmi.get_ceph_image_name_from_project(
+                self.img2, PROJECT)
+            fs.get_image(img_id)
```

Comment:
I am not 100% sure this will work. What if the image is not found ? As far as I know the test won't fail. You will need to fail the test if an exception is raised.

I will recommend testing it first. Try deleting the image before the get_image and see if the test fails. If it does then no issues.

Reply:
Tried and failed. So, there is no issue about it.